packages/sc-provider/src/index.ts

```diff
@@ -32,28 +39,41 @@ const customCreateScClient = (...args: Parameters<typeof createScClient>) => {
 }
 
 let client: ScClient
-
 const noop = () => {}
 
+export type ScProviderConfig = Config & {
+  relayChainSpec?: string
+  client?: ScClient
+}
+
 export const ScProvider = (
   input: WellKnownChain | string,
-  config?: Config,
-): ConnectProvider => {
-  client ??= customCreateScClient(config)
+  config?: ScProviderConfig,
```

Comment:
What if the 2nd parameter is a
- `@substrate/connect` `Config`
- Or, another `ScProvider`

If it is a `ScProvider`, it will reuse the `ScProvider` client and use the `ScProvider` inner chain as `relayChain`

Thoughts?